dataset/imagenet.py

```diff
@@ -0,0 +1,148 @@
+    return im[y : (y + final_size), x : (x + final_size), :]
+
+
+  def _random_sized_crop(self, example, min_area = 0.08, max_iter = 10):
+    h, w = tf.cast(example['height'], tf.int64), tf.cast(example['width'], tf.int64)
+    area = h * w
+    return_example = False
+    image = example['image']
+    num_iter = tf.constant(0, dtype = tf.int32)
+    while num_iter <= max_iter:
+      num_iter  = tf.math.add(num_iter,1)
+      final_area = tf.random.uniform((), minval = min_area, maxval = 1,
+                                      dtype = tf.float32) * tf.cast(area, tf.float32)
+      aspect_ratio = tf.random.uniform((), minval = 3./4., maxval = 4./3.,
+                                      dtype = tf.float32)
+      w_cropped = tf.cast(tf.math.round(tf.math.sqrt(final_area * aspect_ratio)), tf.int64)
+      h_cropped = tf.cast(tf.math.round(tf.math.sqrt(final_area / aspect_ratio)), tf.int64)
+
+      if tf.random.uniform(()) < 0.5:
+        w_cropped, h_cropped = h_cropped, w_cropped
+      
+      if h_cropped <= h and w_cropped <= w:
+        return_example = True
+        if h_cropped == h: 
+          y = tf.constant(0, dtype = tf.int64)
+        else:
+          y = tf.random.uniform((),minval = 0, maxval = h - h_cropped, dtype = tf.int64)
+        if w_cropped == w:
+          x = tf.constant(0, dtype = tf.int64)
+        else:
+          x = tf.random.uniform((),minval = 0, maxval = w - w_cropped, dtype = tf.int64)
+      
+        image = image[y : (y + h_cropped), x : x + w_cropped, :]
+        image = tf.image.resize(image, (self.image_size, self.image_size))
+        break
+    if return_example:
+      return {
+          'image' : image,
+          'height' : self.image_size,
+          'width' : self.image_size,
+          'filename' : example['filename'],
+          'label' : example['label']
+        }
+
+    image = self._scale_and_center_crop(image, h, w, self.image_size, self.image_size)
+    return {
+          'image' : image,
+          'height' : self.image_size,
+          'width' : self.image_size,
+          'filename' : example['filename'],
+          'label' : example['label']
+        }
+  
+  @tf.function
+  def _pca_jitter(self, image):
+      image = tf.cast(image, tf.float32)/255.
+      alpha = tf.random.normal((1,3), mean = 0, stddev = 0.1, dtype = tf.float32)
+      alpha = tf.repeat(alpha, repeats = 3, axis=0)
+      eigen_vals = tf.repeat(ImageNet._EIG_VALS, repeats = 3, axis = 0)
+      rgb = tf.math.reduce_sum(ImageNet._EIG_VECS * alpha * eigen_vals, axis = 1)
+      temp_image = []
+      for i in range(3):
+        temp_image.append(image[:, :, i] + rgb[i])
+      image = tf.stack(temp_image, axis=-1)
+      return image 
+    
+ 
+  def make_dataset(self, ):
+    ds = self._read_tfrecs()
+
+    if self.augment_fn == 'default':
+      ds = ds.map(lambda example: self._random_sized_crop(example))
+
+    else:
+      ds = ds.map(lambda example: self.augment_fn(example))
+    return ds
```

Comment:
If my understanding is correct, the current augmentation pipeline only consists of random resized crops, right? But should that be the case?

Also, instead of naming this script as `imagenet.py` I would recommend adding a bit more description. Something like `imagenet_tfrec_loader.py` since this script seems to contain utilities for parsing the TFRecords and creating a `tf.data.Dataset` out of them.

Reply:
I didn't add it right away because the functions have some small bugs. Actually I intend to have all loading and augmentation code related to the ImageNet dataset in this one file. 
Also, a couple of these functions have many redundant cast ops, so I'll clean it up a bit and then add the functions in the `make_dataset` function.

Reply:
Okay. Would suggest modularizing whenever possible to keep the length and readability of the scripts accessible.

Reply:
Okay